### ISSUE TITLE
AttributeAccessIssue thinks empty arrays in dictionaries cannot use append

### ISSUE THREAD

**Issue by Wreckstation** [DetachHead/basedpyright#1206]
### Description

Minimal example:
```
my_dictionary = { "arbitrary array": []}
my_dictionary["arbitrary array"].append("foobar")

# Diagnostics:
# Cannot access attribute "append" for class "str"
# Attribute "append" is unknown [reportAttributeAccessIssue]
```
Empty arrays inside dictionaries are reported as "str".  In addition, basedpyright says the append() method cannot be used.

**Comment by DetachHead** [comment]
closing as per my previous comment
